pkg/v22/key/key.go

```diff
@@ -302,10 +302,6 @@ func serverDomain(clusterGuestConfig v1alpha1.ClusterGuestConfig, cert certs.Cer
 	return string(cert) + "." + strings.TrimLeft(commonDomain, "\t ."), nil
 }
 
-func TenantBaseDomain(clusterGuestConfig v1alpha1.ClusterGuestConfig) string {
-	return fmt.Sprintf("%s.k8s.%s", ClusterID(clusterGuestConfig), DNSZone(clusterGuestConfig))
-}
-
```

Comment:
The domain was duplicated e.g. `fvu6p.k8s.fvu6p.k8s.ghost.westeurope.azure.gigantic.io`. We can just use `key.DNSZone`.